### PR TITLE
Feature: Make executors and feedbacks easier to use outside of the fuzzing loop (extends #2511)

### PR DESCRIPTION
**Pull request by domenukk**
This PR tries to fix changes since #2511 and merge it

### CODE REVIEW

**Comment by domenukk** [comment]
I still don't understand how I can push into other people's branch...

File: libafl/src/executors/command.rs [hunk 1]
```diff
@@ -162,15 +162,15 @@ where
 ///
 /// Construct a `CommandExecutor` by implementing [`CommandConfigurator`] for a type of your choice and calling [`CommandConfigurator::into_executor`] on it.
 /// Instead, you can use [`CommandExecutor::builder()`] to construct a [`CommandExecutor`] backed by a [`StdCommandConfigurator`].
-pub struct CommandExecutor<OT, S, T> {
+pub struct CommandExecutor<I, OT, S, T> {
```
Comment (tokatoka) [inline libafl/src/executors/command.rs:165]: why this guy takes I?

File: libafl/src/executors/forkserver.rs [hunk 2]
```diff
@@ -674,6 +675,110 @@ where
     pub fn coverage_map_size(&self) -> Option<usize> {
         self.map_size
     }
+
+    #[inline]
+    fn execute_input_tracked(&mut self, state: &mut S, input: &TC::Input) -> Result<ExitKind, Error>
```
Comment (tokatoka) [inline libafl/src/executors/forkserver.rs:680]: why do we need this method?

File: libafl/src/feedbacks/map.rs [hunk 3]
```diff
@@ -424,12 +424,16 @@ where
     fn is_interesting(
         &mut self,
         state: &mut S,
-        manager: &mut EM,
-        input: &I,
+        _manager: &mut EM,
+        _input: &<S as UsesInput>::Input,
```
Comment (tokatoka) [inline libafl/src/feedbacks/map.rs:428]: don't use <S as UsesInput>::Input,
feedbacks is already 99% Use*-free module

File: libafl/src/feedbacks/map.rs [hunk 4]
```diff
@@ -424,12 +424,16 @@ where
...
         observers: &OT,
-        exit_kind: &ExitKind,
-    ) -> Result<bool, Error> {
-        let res = self.is_interesting_default(state, manager, input, observers, exit_kind);
+        _exit_kind: &ExitKind,
+    ) -> Result<bool, Error>
+    where
+        EM: EventFirer<State = S>,
```
Comment (tokatoka) [inline libafl/src/feedbacks/map.rs:433]: do you need this boundary?

File: libafl/src/feedbacks/list.rs [hunk 5]
```diff
@@ -64,6 +64,47 @@ libafl_bolts::impl_serdeany!(
     <u8>,<u16>,<u32>,<u64>,<i8>,<i16>,<i32>,<i64>,<bool>,<char>,<usize>
 );
 
+impl<T> ListFeedback<T>
+where
+    T: Debug + Serialize + Hash + Eq + DeserializeOwned + Default + Copy + 'static,
```
Comment (tokatoka) [inline libafl/src/feedbacks/list.rs:69]: i think here not all of these bounds are necessary.

File: libafl/src/executors/command.rs (hunk 1 above)
Comment (domenukk) [reply libafl/src/executors/command.rs:165]: Yeah

File: libafl/src/executors/forkserver.rs (hunk 2 above)
Comment (domenukk) [reply libafl/src/executors/forkserver.rs:680]: Guess you can side-step the counting that way.

Comment (domenukk) [reply libafl/src/executors/forkserver.rs:680]: Renamed it to `execute_input` and `execute_input_uncounted`

File: libafl/src/feedbacks/list.rs (hunk 5 above)
Comment (domenukk) [reply libafl/src/feedbacks/list.rs:69]: Seems like they are

File: libafl/src/feedbacks/concolic.rs [hunk 6]
```diff
@@ -50,7 +65,7 @@ impl<S> StateInitializer<S> for ConcolicFeedback<'_> {}
 
 impl<EM, I, OT, S> Feedback<EM, I, OT, S> for ConcolicFeedback<'_>
 where
-    OT: MatchName,
+    OT: MatchName + ObserversTuple<I, S>,
```
[the remark this answers is not in the document]
Comment (domenukk) [reply libafl/src/feedbacks/concolic.rs:68]: I do :/

[the remark this answers is not in the document]
Comment (domenukk) [reply libafl/src/feedbacks/concolic.rs:68]: Got rid of one

File: libafl/src/feedbacks/list.rs (hunk 5 above)
Comment (domenukk) [reply libafl/src/feedbacks/list.rs]: (removed all of them and needed to add back one by one)

File: libafl/src/feedbacks/map.rs (hunk 4 above)
Comment (domenukk) [reply libafl/src/feedbacks/map.rs]: Good point

File: libafl/src/feedbacks/map.rs (hunk 3 above)
Comment (domenukk) [reply libafl/src/feedbacks/map.rs:428]: What else? Do I bring back `I`?

Comment (domenukk) [reply libafl/src/feedbacks/map.rs:428]: Changed it... a bit?

Comment (tokatoka) [reply libafl/src/feedbacks/map.rs]: I is fine

File: libafl/src/executors/forkserver.rs [hunk 7]
```diff
@@ -644,6 +644,7 @@ where
     OT: ObserversTuple<S::Input, S>,
     S: UsesInput,
     SP: ShMemProvider,
+    TC: TargetBytesConverter,
```
Comment (tokatoka) [inline libafl/src/executors/forkserver.rs:647]: we dont't need this. we already have S, and is used for ObserversTuple<S::Input, S>. so input is just S::Input.

Comment (tokatoka) [reply libafl/src/executors/forkserver.rs:647]: ah ok it's necessary because this one is now moved out of executor

Comment (domenukk) [reply libafl/src/executors/forkserver.rs:647]: We should add TargetBytesConverter support to command executor eventually

**Comment by domenukk** [comment]
Hmm `fuzzers/structure_aware/forkserver_simple_nautilus` is broken with this dude

**Pull request merged**